RecoTracker/FinalTrackSelectors/plugins/TrackMerger.cc

```diff
@@ -163,6 +142,48 @@ TrackCandidate TrackMerger::merge(const reco::Track &inner, const reco::Track &o
      return ret;
 }
 
+void TrackMerger::addSecondTrackHits(std::vector<const TrackingRecHit *>& hits,
+                                     const reco::Track& outer) const {
+    size_t nHitsFirstTrack = hits.size();
+    for (auto it = outer.recHitsBegin(), ed = outer.recHitsEnd(); it != ed; ++it) {
+        const TrackingRecHit *hit = &**it;
+        DetId id(hit->geographicalId());
+        const auto lay = theTrkTopo->layer(id);
+        bool shared = false;
+        bool valid  = hit->isValid();
+        PRINT << "   subdet " << id.subdetId() << "  layer " << theTrkTopo->layer(id) << " valid " << valid << "   detid: " << id() << std::endl;
+        size_t iHit = 0;
+        for ( auto hit2 :  hits) {
+            ++iHit; if (iHit >  nHitsFirstTrack) break;
+            DetId id2 = hit2->geographicalId();
+            if (id.subdetId() != id2.subdetId()) continue;
+            if (theTrkTopo->layer(id2) != lay) continue;
+            if (hit->sharesInput(hit2, TrackingRecHit::all)) { 
+                PRINT << "        discared as duplicate of other hit" << id() << std::endl;
+                shared = true; break; 
+            }
+            if (hit2->isValid() && !valid) { 
+                PRINT << "        replacing old invalid hit on detid " << id2() << std::endl;
+                hit2 = hit; shared = true; break; 
```

Comment:
(this part was in the old code already.)
Static analyzer points out that ``` hit2 = hit; ``` is a dead assignment.
Based on the PRINT content, it looks like the code should be ``` hit = hit2; ```

Please check and follow up in a separate PR.